### PR TITLE
[pvr] add defines for invalid season and episode values

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Add defines to aid in readability and understanding of PVR addons to indicate when episode/season/part values are invalid.

## Motivation and Context

Related to changes made in https://github.com/xbmc/xbmc/pull/17488 and https://github.com/xbmc/xbmc/pull/17408.

## How Has This Been Tested?

No impact to kodi as values only used by addons.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
@ksooo would like to get this in before I make all the PVR addon changes. If you have time for quick review that would be great.

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h [hunk 1]
```diff
@@ -527,15 +527,20 @@ extern "C" {
 
   } ATTRIBUTE_PACKED PVR_TIMER;
 
+  /*!
+   * @brief special PVR_RECORDING.iSeriesNumber and PVR_RECORDING.iEpisodeNumber value to indicate it is not to be used
+   */
+  const unsigned int PVR_RECORDING_INVALID_SERIES_EPISODE = EPG_TAG_INVALID_SERIES_EPISODE; /*!< @brief recording has no valid season and/or episode info. */
```
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h]: About them with docs becomes updated on my request, but it make not directly sense about doxygen to have description before and behind.

Further where also a bug on the other places, does doxygen not use a `/*!<` a with `@brief` defined text. Only without this `@brief`

Maybe better to make directly the doc together:
```cpp
  /*!
   * @brief special PVR_RECORDING.iSeriesNumber and PVR_RECORDING.iEpisodeNumber
   *  value to indicate it is not to be used
   *
   *  Used if recording has no valid season and/or episode info.
   */
  const unsigned int PVR_RECORDING_INVALID_SERIES_EPISODE = EPG_TAG_INVALID_SERIES_EPISODE;```

Comment (phunkyfish) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h]: Ok, cool. Done ;)

Comment (AlwinEsch) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h]: There was on begin wrongly by c&p two characters on `*  Used if` see edited text.

Comment (phunkyfish) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_pvr_types.h]: Ok, fixed now ;)

**Comment by lrusak** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
@AlwinEsch are you happy with this? I would like to push it in later today.

**Review by ksooo — APPROVED** [review]

**Pull request merged**